core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1552,13 +1552,15 @@ private void completeFileInternal(RpcContext rpcContext, LockedInodePath inodePa
 
     // We could introduce a concept of composite entries, so that these two entries could
     // be applied in a single call to applyAndJournal.
-    mInodeTree.updateInode(rpcContext, UpdateInodeEntry.newBuilder()
+    UpdateInodeEntry.Builder builder = UpdateInodeEntry.newBuilder()
         .setId(inode.getId())
-        .setUfsFingerprint(ufsFingerprint)
-        .setLastModificationTimeMs(opTimeMs)
-        .setLastAccessTimeMs(opTimeMs)
-        .setOverwriteModificationTime(true)
-        .build());
+        .setUfsFingerprint(ufsFingerprint);
+    if (ServerConfiguration.getBoolean(PropertyKey
+        .MASTER_FILE_SYSTEM_FINAL_TIME_UPDATE_ENABLED)) {
+      builder.setLastModificationTimeMs(opTimeMs)
```

Comment:
@bzheng888 did Alluxio update mtime and atime when creating the file and closing the file?
we set the values twice?

Reply:
@LuQQiu Yes, On the master side, createFile will call InodeTree.createPath->MutableInodeFile.create, this will set mtime and ctime the first time, then the completeFile will set again.